source.go

```diff
@@ -25,6 +25,8 @@ type Source interface {
 	SetReceiveHandler(fn func(tag frame.Tag, data []byte))
 	// Write the data to all downstream
 	Broadcast(data []byte) error
+	// NewStream will create a new data stream [experimental feature]
+	NewStream(metadata []byte) (core.DataStream, error)
```

Comment:
Shall we consider provide `.WriteStream(tag, readableStream)` instead of `NewStream()` as YoMo is a framework, w'd like to discuss more and make some usage tests on this topic.